storage/integration_test.go

```diff
@@ -815,13 +815,16 @@ func TestIntegration_PublicAccessPrevention(t *testing.T) {
 
 		// Now, making object public or making bucket public should succeed. Run with
 		// retry because ACL settings may take time to propagate.
-		idempotentOrNilRetry := func(err error) bool {
-			return err == nil || ShouldRetry(err)
+		retrier := func(err error) bool {
+			// Once ACL settings propagate, PAP should no longer be enforced and the call will succeed.
+			// In the meantime, while PAP is enforced, trying to set ACL results in:
+			// 	-	FailedPrecondition for gRPC
+			// 	-	condition not met (412) for HTTP
+			return ShouldRetry(err) || status.Code(err) == codes.FailedPrecondition || extractErrCode(err) == 412
```

Comment:
Nit: Consider using `http.StatusPreconditionFailed` instead of the magic number `412`.

Reply:
good one 👍